### ISSUE TITLE
Highlighting in nested queries

### ISSUE THREAD

**Issue by joblolabinette** [dbeaver/dbeaver#22715]
### Description

Writing a nested query causes problems with the new highlighting (which I should say is very nice!)
Whereas using the CTE structure works well. In the case where there is just a variable, everything is gray. When there is more than 1 the 2nd and subsequent are red.

![Queries](https://github.com/dbeaver/dbeaver/assets/3724319/d04674cc-6be2-4c8b-99f9-73da246adf55)

### DBeaver Version

CE 23.3.4

### Operating System

Windows

### Database and driver

PostgreSQL 14.7.0
PostgreSQL JDBC Driver 42.2.5

**Comment by E1izabeth** [comment]
Thank you for bug report.

**Comment by joblolabinette** [comment]
A simpler example, if it can help:
![Queries2](https://github.com/dbeaver/dbeaver/assets/3724319/62aa8585-060e-4350-aac3-f1acf650ec8c)

**Comment by E1izabeth** [comment]
@joblolabinette it doesn't look like a correct query actually. When I try executing a similar one, I get this
![image](https://github.com/dbeaver/dbeaver/assets/28875055/0acb69ef-34ec-4833-8040-400dd6635779)

**Comment by joblolabinette** [comment]
I added an alias and the highlighting works. 
However it looks like Postgresql engine doesn't require an alias. The query returns without problem.

**Comment by joblolabinette** [comment]
I added an alias and the highlighting works. 

We are using PostgreSQL 14.7 (EnterpriseDB Advanced Server 14.7.0) if it changes anything over vanilla PostgreSQL. The query returns without problem.
I also found this online. https://pganalyze.com/blog/5mins-postgres-waiting-for-postgres-16-subquery-alias-optional

Thanks for taking the time to look into this. But for now I know adding an alias fixes the highlighting.
